packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.types.ts

```diff
@@ -288,6 +288,9 @@ export interface IGroupDividerProps {
 
   /** Props for expand/collapse button */
   expandButtonProps?: React.HTMLAttributes<HTMLButtonElement>;
+
```

Comment:
Just do:
```ts
groups?: IGroup[]; // or 'allGroups'
```
This makes it simple for the caller to get access to its siblings (and their data, if needed). I'd avoid removing the current group itself because that way a group can find itself among its siblings if needed.